src/utils/source-nodes.js

```diff
@@ -37,7 +37,17 @@ export async function sourceNodes(
     createNode(hashNodeUser);
 
     // get posts
-    const posts = await getUserPosts(username);
+    let page = 0;
+    let posts = [];
+    while (true) {
```

Comment:
Can you please find a solutio which doesn't involves infinite loops but gets the number
of pages from hashnode's api and iterates that.

If not possible, iterate a big number (9999) until there are no pages

Reply:
Unfortunately, the API doesn't seem to have a field for pages and `numPosts` on the user always returns `null`. I have put in a for loop with 9999 as you suggested. The break is still in there so it will still exit once there are no more posts.